### PR TITLE
Update wesnoth-utbs.po

### PR DESCRIPTION
**Pull request by MultiSeptim**
ke commit b810cc6

### CODE REVIEW

File: wesnoth-utbs.po [hunk 1]
```diff
@@ -8806,16 +8744,12 @@ msgstr ""
...
-#| msgid ""
-#| "Nog no like humans either. They mean. But they sound great when they go "
-#| "squish."
 msgid ""
 "$ally_unit.name no like humans either. They mean. But they sound great when "
 "they go squish."
 msgstr ""
-"Nog taky nemá rád lidi. Jsou skoupí. Ale dělají skvělé zvuky když se "
-"rozmačkávají."
+"$ally_unit.name taky nemá rád lidi. Jsou skoupí. Ale dělají skvělé zvuky "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"$ally_unit.name taky nemá rád lidi. Jsou skoupí. Ale dělají skvělé zvuky, "
```

File: wesnoth-utbs.po [hunk 2]
```diff
@@ -11196,17 +11121,14 @@ msgstr ""
...
-#| "Leader. Great Leader say follow Kaleh, and Nog will do so."
 msgid ""
 "$ally_unit.name scared of big water and bright sun, but $ally_unit.name will "
 "not dishonor Great Leader. Great Leader say follow Kaleh, and $ally_unit."
 "name will do so."
 msgstr ""
-"Nog bojí velké vody a ostrého slunce, ale Nog nezahanbí Velkého vůdce. Velký "
-"vůdce řekl jdi s Kalehem, a Nog to udělá."
+"$ally_unit.name bojí velké vody a ostrého slunce, ale $ally_unit.name "
+"nezahanbí Velkého vůdce. Velký vůdce řekl jdi s Kalehem, a $ally_unit.name "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"nezahanbí Velkého vůdce. Velký vůdce řekl jdi s Kalehem a $ally_unit.name "
```

File: wesnoth-utbs.po [hunk 3]
```diff
@@ -8313,10 +8254,11 @@ msgid ""
 "caves of his people. But for now, $ally_unit.name will continue to serve and "
 "protect you."
 msgstr ""
-"Ale Nog se nebojí. Velký vůdce řekl, Nog vést a chránit elfy, a Nog splní "
-"svůj slib. Nog bude následovat elfy kamkoliv půjdou a chránit je před "
-"nebezpečím, jak jen to půjde. Třeba později najde jinou cestu zpátky do "
-"jeskyní ke trollům. Ale zatím vám bude Nog dál sloužit a chránit vás."
+"Ale $ally_unit.name se nebojí. Velký vůdce řekl, $ally_unit.name vést a "
+"chránit elfy, a $ally_unit.name splní svůj slib. $ally_unit.name bude "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"Ale $ally_unit.name se nebojí. Velký vůdce řekl $ally_unit.name vést a "
"chránit elfy a $ally_unit.name splní svůj slib. $ally_unit.name bude "
```

File: wesnoth-utbs.po [hunk 4]
```diff
@@ -10601,19 +10531,14 @@ msgstr ""
...
-#| "cross deep water?"
 msgid ""
 "Great Leader told $ally_unit.name to serve you, and so $ally_unit.name will "
 "still follow your command. But other elves must be out on a separate island. "
 "How will we cross deep water?"
 msgstr ""
-"Velký vůdce řekl, Nog bude sloužit Kalehovi, a Nog bude stále poslouchat tvé "
-"rozkazy. Ale druzí elfové musí být na dalším ostrově. Jak překročíme "
-"hlubokou vodu?"
+"Velký vůdce řekl, $ally_unit.name bude sloužit Kalehovi, a $ally_unit.name "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"Velký vůdce řekl $ally_unit.name bude sloužit Kalehovi a $ally_unit.name "
```

File: wesnoth-utbs.po [hunk 5]
```diff
@@ -11819,16 +11734,12 @@ msgstr ""
...
-#| msgid ""
-#| "Nog not sure if he trust fish lady. What if she just want you elves to "
-#| "fight her war for her?"
 msgid ""
 "$ally_unit.name not sure if he trust fish lady. What if she just want you "
 "elves to fight her war for her?"
 msgstr ""
-"Nog neví jestli věří rybí paní. Co když chce aby elfové bojovali za ni v "
-"její válce?"
+"$ally_unit.name neví jestli věří rybí paní. Co když chce aby elfové bojovali "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"$ally_unit.name neví, jestli věří rybí paní. Co když chce, aby elfové bojovali "
```

File: wesnoth-utbs.po [hunk 6]
```diff
@@ -12016,8 +11921,8 @@ msgid ""
 msgstr ""
 "Ani v nejhlubších tunelech můj lid neunikne silám Yechnagoth navždy, jestli "
 "ji nikdo nezastaví. Proti takovému zlu by vám určitě dal Velký vůdce celou "
-"armádu trollů, ale teď je tu jenom Nog. I tak Nog udělá všechno aby ji "
-"zničil."
+"armádu trollů, ale teď je tu jenom $ally_unit.name. I tak $ally_unit.name "
+"udělá všechno aby ji zničil."
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"udělá všechno, aby ji zničil."
```

File: wesnoth-utbs.po [hunk 7]
```diff
@@ -13706,14 +13580,15 @@ msgid ""
...
-"na vody, na svět, kterému jsme tolik obětovali, abychom jej uchovali. "
-"Připomínám si, co říkala Zhul, a čemu všichni věřili, totiž že tento svět je "
-"navzdory všem bojům a smrti, které jsme zažili během naší cesty natolik "
-"pěkné místo, že Jarl neváhal obětovat svůj život, aby jej zachránil. Jak "
-"hledím přes vody a připomínám si blahobyt, v němž nyní žije můj lid, opakuji "
-"si, že to za to stálo. Ale někdy je to malá útěcha."
+"stále mě děsí vzpomínka na to, jak na tom temném místě zemřel $ally_unit."
+"name. A tak přicházím každé ráno po východu slunce na jihovýchodní cíp "
+"ostrova a hledím na vody, na svět, kterému jsme tolik obětovali, abychom jej "
+"uchovali. Připomínám si, co říkala Zhul, a čemu všichni věřili, totiž že "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"uchovali. Připomínám si, co říkala Zhul a čemu všichni věřili, totiž že "
```

File: wesnoth-utbs.po [hunk 8]
```diff
@@ -13706,14 +13580,15 @@ msgid ""
...
+"tento svět je navzdory všem bojům a smrti, které jsme zažili během naší "
+"cesty natolik pěkné místo, že $ally_unit.name neváhal obětovat svůj život, "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"cesty, natolik pěkné místo, že $ally_unit.name neváhal obětovat svůj život, "
```

File: wesnoth-utbs.po [hunk 9]
```diff
@@ -13831,15 +13694,16 @@ msgid ""
...
-"skaliscích, několik menších jeskyní. Nebyly tak hluboké, jako v jeho domově, "
-"ale myslím, že Jarl by ocenil alespoň tu trochu solidního kamene nad místem "
-"jeho posledního odpočinku. Tak jsme jej tedy pohřbili se všemi poctami. "
-"Dlouho budeme vzpomínat na jeho službu, kterou nám prokázal on a jeho lid."
+"Smrt mého přítele z podzemí mě zarmoutila, ale zachránil mi život a tak "
+"naplnil svůj dluh. Myslím, že je rád, že padl v bitvě. Uvažoval jsem o "
+"vyslání expedice, která by vrátila jeho tělo jeho lidu, ale všichni jsme "
+"byli vysíleni strázněmi dlouhé cesty a nechtěl jsem zbytečně riskovat. Místo "
+"toho jsem po celém ostrově hledal příhodné místo. Nakonec jsem nalezl v "
+"severozápadních skaliscích několik menších jeskyní. Nebyly tak hluboké, jako "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"severozápadních skaliscích několik menších jeskyní. Nebyly tak hluboké jako "
```

File: wesnoth-utbs.po [hunk 10]
```diff
@@ -13926,37 +13774,22 @@ msgid ""
...
-"mi, že se musí vrátit ke svému lidu. Čas, který s námi strávil, mu podle "
-"jeho slov připadal jako krásný sen; slíbil, že na nás nikdy nezapomene, ale "
-"jeho lid jej potřeboval a musel se vrátit. Prý se jednou vrátí a znovu nás "
-"navštíví, ale pochybuji, že jej ještě kdy uvidím. I já si budu navždy "
-"pamatovat jej i jeho druh, a dlouho budu vzpomínat na jeho neotřesitelnou "
-"důvěru a vše, čím nám v našich strastech pomohl."
+"Záchranou mého života naplnil $ally_unit.name svůj dluh života, ale "
+"přesvědčili jsme jej, aby s námi ještě chvíli zůstal a veselil se. Po našem "
+"úžasném vítězství jsme celé dny slavili, děkujíc Eloh a bohu mořského lidu a "
+"radujíc se z krásy a bohatství našeho nového domova. Poté jsme se dali do "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"úžasném vítězství jsme celé dny slavili, děkujíce Eloh a bohu mořského lidu a "
"radujíce se z krásy a bohatství našeho nového domova. Poté jsme se dali do "
```

File: wesnoth-utbs.po [hunk 11]
```diff
@@ -13926,37 +13774,22 @@ msgid ""
...
+"práce a začali si budovat nové příbytky. $ally_unit.name s námi ještě zůstal "
+"a pomáhal při stavbě, byl z nás všech nejpilnější stavitel. Ale po nějakém "
+"čase za mnou přišel a řekl mi, že se musí vrátit k svým druhům. Čas, který s "
+"námi strávil, mu podle jeho slov připadal jako krásný sen, slíbil, že na nás "
+"nikdy nezapomene. Ale jeho druhové jej potřebovali a musel se za nimi "
+"navrátit. Prý se jednou vrátí a znovu nás navštíví, ale pochybuji, že jej "
+"ještě kdy uvidím. I já si budu navždy pamatovat jej i jeho druh, a dlouho "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"ještě kdy uvidím. I já si budu navždy pamatovat jej i jeho druh a dlouho "
```

File: wesnoth-utbs.po [hunk 12]
```diff
@@ -13973,35 +13806,31 @@ msgid ""
...
-"mi, že se musí vrátit ke svému lidu. Čas, který s námi strávil, mu podle "
-"jeho slov připadal jako krásný sen; slíbil, že na nás nikdy nezapomene, ale "
-"jeho lid jej potřeboval a musel se vrátit. Prý se jednou vrátí a znovu nás "
-"navštíví, ale pochybuji, že jej ještě kdy uvidím. I já si budu navždy "
-"pamatovat jej i jeho druh, a dlouho budu vzpomínat na jeho neotřesitelnou "
-"důvěru a vše, čím nám v našich strastech pomohl."
+"Záchranou mého života naplnil $ally_unit.name svůj dluh života, ale "
+"přesvědčili jsme jej, aby s námi ještě chvíli zůstal a veselil se. Po našem "
+"úžasném vítězství jsme celé dny slavili, děkujíc Eloh a bohu mořského lidu a "
+"radujíc se z krásy a bohatství našeho nového domova. A poté jsme se dali do "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"úžasném vítězství jsme celé dny slavili, děkujíce Eloh a bohu mořského lidu a "
"radujíce se z krásy a bohatství našeho nového domova. A poté jsme se dali do "
```

File: wesnoth-utbs.po [hunk 13]
```diff
@@ -13973,35 +13806,31 @@ msgid ""
...
+"práce a začali si budovat nové příbytky. $ally_unit.name s námi ještě zůstal "
+"a pomáhal při stavbě, byl z nás všech nejpilnější stavitel. Ale po nějakém "
+"čase za mnou přišel a řekl mi, že se musí vrátit k svým druhům. Čas, který s "
+"námi strávil, mu podle jeho slov připadal jako krásný sen, slíbil, že na nás "
+"nikdy nezapomene. Ale jeho druhové jej potřebovali a musel se za nimi "
+"navrátit. Prý se jednou vrátí a znovu nás navštíví, ale pochybuji, že jej "
+"ještě kdy uvidím. I já si budu navždy pamatovat jej i jeho druh, a dlouho "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"ještě kdy uvidím. I já si budu navždy pamatovat jej i jeho druh a dlouho "
```

File: wesnoth-utbs.po [hunk 14]
```diff
@@ -13973,35 +13806,31 @@ msgid ""
...
-#| "glad for the short time that I knew him."
 msgid ""
 "And I will always remember $ally_unit.name who died along our journey. A "
 "braver warrior I have never seen, and though he was taken from us too soon, "
 "I am glad for the short time that I knew him."
 msgstr ""
-"Stále budu vzpomínat i na Jarla, který zemřel během naší cesty. Nikdy jsem "
-"neviděl statečnějšího válečníka, a i když nás opustil příliš brzy, jsem rád, "
-"že jsem jej znal alespoň po ten krátký čas."
+"Stále budu vzpomínat i na mého přítelem jménem $ally_unit.name, který zemřel "
```
Comment (celerini) [inline wesnoth-utbs.po]:
```suggestion
"Stále budu vzpomínat i na mého přítele jménem $ally_unit.name, který zemřel "
```

**Review by celerini — CHANGES_REQUESTED** [review]

**Comment by MultiSeptim** [comment]
Obecně ale nejsem nadšen těmito typy změn. Dávat do textů jména jako proměnou vždy komplikuje překlady. Chtěl bych vidět, jak se s tím popasují jazyky, které nepoužívají latinku. To bude jako v textu v azbuce jméno latinkou?

**Comment by celerini** [comment]
It may be worth to explain our translation difficulties (as I [tried earlier](https://github.com/hrubymar10/wesnoth-cs/issues/170#issuecomment-903303312)) to the author of this [change](https://github.com/wesnoth/wesnoth/issues/6158) and ask him directly.

**Comment by Wedge009** [comment]
What's the issue here? With respect to https://github.com/wesnoth/wesnoth/pull/6165, which involved @stevecotton, @CelticMinstrel, and myself, the intention was simply to remove the redundant duplication of strings. However simple it might be translate duplicate strings, I think it best to avoid them in the source for similar reasons that we avoid redundant duplication in code. Is the issue solely to do with variables that are included within the translated text or is it something more?

**Comment by MultiSeptim** [comment]
Yes it does. Strings with variables may be difficult to translate.
I reacted to the latest changes in UtBS (main). I dealt with them, but it wasn't easy. Somewhere I had to omit those names from the translation, for example, I used phrase _a dwarf/troll friend_ instead of $ally_unit.name.
I agre that code should be clean. But I haven't registered that $ally_unit.name could be varied somewhere yet (e.g. due to declension).

**Comment by Wedge009** [comment]
If I recall correctly, it can be tricky for troll speech because in English speaking in the third person can be interpreted as a sign of low intelligence, and trolls are often characterised this way. Hence sentences like 'Grog promised to help little dwarves' - but 'Grog' is just replaced with the variable name instead (because the troll could be replaced with one of another name).

I'm neither a linguist nor multi-lingual, so pardon me for not fully understanding but is there something more complex to consider than just substituting names for the variable names when reading the lines for localisation? I should add that this isn't a problem unique to the refactoring - it could potentially be present in other campaigns where variables are used.

**Comment by MultiSeptim** [comment]
It's not so "complicated", it's just how to handle variables well for translation. And yes, this is a very general problem. We figured it, for example, in [WC](https://github.com/wesnoth/wesnoth/issues/5114).

**Comment by celerini** [comment]
We (Czech) are changing (mainly by adding a suffix) nouns (adjectives, pronouns...) according to their gender, grammatical paradigm (pattern, type) and grammatical case (context). Although the case (context) is known from surrounding text, both the gender and the paradigm is not known when using variables. So we can't append the correct suffix to the variable and thus all such translations looks dumb as if it would be a troll speach.

English is simpler than Czech (thus you can deduplicate a text in same manner as a code), but has its own remains of the grammatical case. Consider **he** - I see **he** (not **him**) may look similar in English as when we use a plain variable name in Czech.

I think it can be overcome by using generators, but not for a sake of simplicity. Please se [my previous comment](https://github.com/hrubymar10/wesnoth-cs/issues/170#issuecomment-903303312).

Is it more clear now?

**Comment by Wedge009** [comment]
For the specific case of UtBS, where these dwarf and troll allies are concerned, they are always male.

Edit: I see this example of pre-existing use of a variable name. Does the translation simply drop the reference? (In English, that would be valid.)
https://github.com/hrubymar10/wesnoth-cs/blob/a543abdc29574383db652edaa82bc7ea02760fff/wesnoth-dw.po#L2381-L2388

Edit 2: This translation appears to show no difference between a reference to a male unit and a female unit - is that correct?
https://github.com/hrubymar10/wesnoth-cs/blob/a543abdc29574383db652edaa82bc7ea02760fff/wesnoth-httt.po#L7291-L7299

**Comment by CelticMinstrel** [comment]
I get the issue. It's easy to forget that nouns (even proper nouns) decline in some languages, and I don't even have the excuse of not being a linguist or multilingual. Although it's unfortunate, given the effort @Wedge009 and I put into that pull request, the simplest solution to this problem is likely to just revert it. Looking at it, as far as I can tell it didn't include any other changes beyond the string deduplication.

That said, this pull request (which seems unrelated?) probably isn't the best place to discuss it. I might've missed the whole conversation if I hadn't misclicked my notifications.

**Comment by MultiSeptim** [comment]
In the first example, the name (noun) is in the fifth case. So variable will not work properly.
In the second example, the name (noun) is in the first case. So variable could work (male and female too), except the situation when we translate the name also (like Conrad -> Konrád).

**Comment by Wedge009** [comment]
> That said, this pull request (which seems unrelated?) probably isn't the best place to discuss it. I might've missed the whole conversation if I hadn't misclicked my notifications.

I replied here, because the question was put to me directly in the Wesnoth issue report (not the PR) relating to the text consolidation. Are you saying that you're in favour of reverting that work? I'm uncomfortable with having all that duplication for the reasons we worked on the PR in the first place, plus I was reviewing the changes to see if po hints could be added to assist.

> In the first example, the name is in the fifth case. So variable will not work properly. In the second example, the name is in the first case. So variable could work (male and female too), except the situation when we translate the name also (like Conrad -> Konrád).

I appreciate that. The point of the examples I listed was to point out that the challenges you're facing with variables is a general translation difficulty, not specific to the UtBS consolidation we did.

**Comment by CelticMinstrel** [comment]
> In the first example, the name (noun) is in the fifth case. So variable will not work properly.
> In the second example, the name (noun) is in the first case. So variable could work (male and female too), except the situation when we translate the name also (like Conrad -> Konrád).

Even if that's true for one language, I don't think I can guarantee it's true for every supported language.

> Are you saying that you're in favour of reverting that work? I'm uncomfortable with having all that duplication for the reasons we worked on the PR in the first place, plus I was reviewing the changes to see if po hints could be added to assist.

I don't like seeing duplication either, but as this exact issue shows, deduplication can hurt localization. I think a localization system could theoretically be made that handles this automatically, but gettext is not that (and I don't believe one exists), so unless you can come up with a better idea, then… yes, I am reluctantly in favour of reverting it.

**Comment by Wedge009** [comment]
Have reverted the text consolidation and re-opened the associated issue.

**Comment by CelticMinstrel** [comment]
The associated issue is wesnoth/wesnoth#6158.

**Pull request merged**